compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -8220,9 +8208,34 @@ private STNode parseCallStatement(STNode expression) {
         // This is not a must because this expression is validated in the semantic analyzer.
         STNode semicolon = parseSemicolon();
         endContext();
+        if (expression.kind == SyntaxKind.CHECK_EXPRESSION) {
+            STNode callExpr = parseCallExpression((STCheckExpressionNode) expression);
```

Comment:
lets assign this to `expression`
`validateCallExpression` is appropriate than `parseCallExpression`
lets do the casting inside the method